marshmallow/fields.py

```diff
@@ -915,7 +916,6 @@ def _deserialize(self, value, attr, data):
         """Deserialize an ISO8601-formatted time to a :class:`datetime.time` object."""
         if not value:   # falsy values are invalid
             self.fail('invalid')
-            raise err
```

Comment:
Yeah, this could be removed in an act of bureaucracy and properness :) I'm more in favor of the [Boy Scout rule](bureaucracy). But if it helps merging this PR, sure, I can undo all cleanup. (By the way, I was not looking for things to clean up, I just stumbled over them.)